services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -65,6 +67,7 @@ public class RouterHttpRequestStats extends AbstractVeniceHttpStats {
   private final Sensor multiGetFallbackSensor;
   private final Sensor metaStoreShadowReadSensor;
   private Sensor keySizeSensor;
+  private Sensor systemStoreSensor = null;
```

Comment:
The approach of the singleton sensor only works if it's going to be a dummy sensor, but in order to consolidate all system stores of a given type together, we can't do it with a singleton sensor. Each metric still needs a different sensor, even though it's going to be the same sensor as for other system stores of the same type.

What I would suggest is this:

1. Instead of a singleton sensor, let's have a `private final String systemStoreName` here.
2. In the constructor, let's do something like: `this.systemStoreName = VeniceSystemStoreUtils.isSystemStore(storeName) ? VeniceSystemStoreUtils.extractSystemStoreType(storeName) : null;`
3. Then in `registerSensor`, let's do something like `return super.registerSensor(getFullMetricName(this.systemStoreName == null ? sensorName : this.systemStoreName), null, stats);`

In this way, each metric will have its own sensor, but it will always be the same sensor for a given system store type.

Reply:
Thanks for the review., there another bug, it was using `super.registerSensor(getFullMetricName(this.systemStoreName == null ? ....), null, stats)` as `super.registerSensor` will again call `getFullMetricName` on the already full metric name.